hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/AbstractInferableTypeSqmExpression.java

```diff
@@ -42,15 +42,16 @@ public ExpressableType getExpressableType() {
 	@Override
 	public Supplier<? extends ExpressableType> getInferableType() {
 		return () -> {
-			final ExpressableType inferredType;
-			if ( typeInference != null ) {
-				inferredType = typeInference.get();
-			}
-			else {
-				inferredType = null;
-			}
-
-			return firstNonStandardBasicType( inferredType, inherentType );
+//			final ExpressableType inferredType;
+//			if ( typeInference != null ) {
+//				inferredType = typeInference.get();
+//			}
+//			else {
+//				inferredType = null;
+//			}
+//
+//			return firstNonStandardBasicType( inferredType, inherentType );
+			return inherentType;
 		};
 	}
 
```

Comment:
may be we can get rid of of  `getInferableType() ` and  just use the `getExpressableType()` of the `InferableTypeSqmExpression` .